morp/interface/__init__.py

```diff
@@ -40,13 +41,82 @@ def set_interface(interface, connection_name=""):
 
 class InterfaceMessage(object):
     """this is a thin wrapper around all received interface messages"""
-    def __init__(self, fields, raw_msg=None):
+
+    @property
+    def body(self):
+        d = self.depart()
+        return self._encode(d)
+
+    @body.setter
+    def body(self, b):
+        d = self._decode(b)
+        self.update(fields=d)
+
+    def __init__(self, interface, raw):
         """
-        fields -- dict -- the original fields you passed to the Interface send method
-        raw_msg -- mixed -- this is the raw message the interface returned
+        interface -- Interface -- the specific interface to send/receive messages
+        raw -- mixed -- the raw message the interface returned
         """
-        self.fields = fields
-        self.raw_msg = raw_msg
+        self.fields = {} # the original fields you passed to the Interface send method
+        self.interface = interface
+        self.raw = raw
+        self.update()
+
+    def depart(self):
```

Comment:
This is a weird name. I've never seen a 'depart' method before, and I'm not sure what it's supposed to represent.